src/plugins/explore/public/application/pages/traces/trace_details/state/trace_app_state.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  createStateContainer,
+  syncState,
+  IOsdUrlStateStorage,
+} from '../../../../../../../opensearch_dashboards_utils/public';
+
+export interface TraceAppState {
+  traceId: string;
+  dataSourceId: string;
+  indexPattern: string;
+  spanId?: string;
+}
+
+const STATE_STORAGE_KEY = '_a';
+
+export const createTraceAppState = ({
+  stateDefaults,
+  osdUrlStateStorage,
+}: {
+  stateDefaults: TraceAppState;
+  osdUrlStateStorage: IOsdUrlStateStorage;
+}) => {
+  const initialStateFromUrl = osdUrlStateStorage.get<TraceAppState>(STATE_STORAGE_KEY);
+  const initialState: TraceAppState = {
+    ...stateDefaults,
+    ...initialStateFromUrl,
+  };
+
+  const stateContainer = createStateContainer(initialState, {
```

Comment:
This file implements a separate state management system from the main explore plugin's Redux store. Have we discussed somewhere about the state management for different pages? Currently explore uses Redux + persistence middleware for URL sync. I am curious why we create two different state management patterns within the same plugin. Should we consider integrating with existing Redux store?

Reply:
The trace details page uses a separate state management system because it serves a fundamentally different purpose than the main explore interface. The main Redux store manages complex query building, visualization configurations, and editor modes - none of which are relevant to trace navigation. 

The trace details page only needs to manage trace-specific navigation state (traceId, spanId, dataSourceId, indexPattern). Using a lightweight state container provides better performance and cleaner URLs while maintaining the same URL synchronization pattern.

This architectural separation follows the principle of using the right tool for the job - complex Redux for complex query workflows, simple state containers for simple navigation state.

Reply:
could you give an example of an URL for the detail page? And a example on a non-detail trace page? I'm going back and forth in writing a comment here, but I think examples will help

Reply:
Discover:Trace:
http://localhost:5601/zky/w/ah0CZq/app/explore/traces/#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now))


Trace Details:
http://localhost:5601/lvy/w/9C3CHg/app/explore/traces/traceDetails#/?_a=(dataSourceId:e2681970-5b5b-11f0-84cd-1f27cd4c7d84,indexPattern:'otel-v1-apm-span-*',spanId:a8b6dd34808083b3,traceId:'23a88fa559da1b6c7521658fba2cb82f')